drivers/sensors/pimoroni_trackball.h

```diff
@@ -22,8 +22,21 @@
 #ifndef TRACKBALL_ADDRESS
 #    define TRACKBALL_ADDRESS 0x0A
 #endif
-#define TRACKBALL_WRITE ((TRACKBALL_ADDRESS << 1) | I2C_WRITE)
-#define TRACKBALL_READ (TRACKBALL_ADDRESS << 1)
+
+#ifndef TRACKBALL_INTERVAL_MS
+#    define TRACKBALL_INTERVAL_MS 8
+#endif
+
+#ifndef TRACKBALL_DEBOUNCE_ONCLICK
+#    define TRACKBALL_DEBOUNCE_ONCLICK 5
+#endif
+typedef struct pimoroni_data {
+    uint8_t left;
+    uint8_t right;
+    uint8_t up;
+    uint8_t down;
+    uint8_t click;
+} pimoroni_data;
```

Comment:
From a quick scan, I dont see intent in the API to make `TRACKBALL_ADDRESS`, `TRACKBALL_INTERVAL_MS`, `TRACKBALL_DEBOUNCE_ONCLICK` or `pimoroni_data` public. It should be safe to move them to the top of the `.c` file.

Reply:
Thank, noticed `trackball_register_button` isn't actually a thing in the old driver either, so removed it.